### PR TITLE
Mo 82 add validation for day working hours in custom timesheets

### PR DESCRIPTION
**Pull request by RajatMCAF**
Added the condition to check 24 hrs for one day that it should not exceed this limit and if exceedes then error message pops up

### CODE REVIEW

File: mo_hr_timesheets/models/custom_grid_timesheet.py [hunk 1]
```diff
@@ -1,53 +1,87 @@
...
+            existing_hours_for_task = sum(
+                self.search([('date', '=', date),
+                             ('task_id', '=', task_id)]).mapped('unit_amount'))
+            total_hours_for_day = (total_hours_for_day -
+                                   existing_hours_for_task +
+                                   change)
+        else:
+            total_hours_for_day += change
+
+        if total_hours_for_day > 24:
```
Comment (MORPSmcaf) [inline mo_hr_timesheets/models/custom_grid_timesheet.py:77]: change magic number to constant

File: mo_hr_timesheets/models/custom_grid_timesheet.py [hunk 2]
```diff
@@ -1,53 +1,87 @@
...
+            raise ValidationError(_("Total hours for date %(date)s "
```
Comment (MORPSmcaf) [inline mo_hr_timesheets/models/custom_grid_timesheet.py:78]: works untill 12 hous, check and change message if you need

File: mo_hr_timesheets/models/custom_grid_timesheet.py (hunk 1 above)
Comment (RajatMCAF) [reply mo_hr_timesheets/models/custom_grid_timesheet.py:77]: Done

File: mo_hr_timesheets/models/custom_grid_timesheet.py (hunk 2 above)
Comment (RajatMCAF) [reply mo_hr_timesheets/models/custom_grid_timesheet.py:78]: Done

File: mo_hr_timesheets/models/custom_grid_timesheet.py [hunk 3]
```diff
@@ -1,53 +1,87 @@
...
             if isinstance(domain_part, (list, tuple)):
                 if domain_part[0] == 'project_id' and domain_part[1] == '=':
                     project_id = domain_part[2]
                 elif domain_part[0] == 'task_id' and domain_part[1] == '=':
                     task_id = domain_part[2]
 
-        if not project_id or not task_id:
-            return False
+        if not project_id:
+            raise ValidationError(_("Project ID is required "
```
Comment (MORPSmcaf) [inline mo_hr_timesheets/models/custom_grid_timesheet.py:52]: Do we need "ID" in error message or "Project" is enough. To avoid confusing users

Comment (RajatMCAF) [reply mo_hr_timesheets/models/custom_grid_timesheet.py]: I actually put this validation for testing purpose this statement will be true 99%. So I removed the validation and changed it to return False

File: mo_hr_timesheets/models/custom_grid_timesheet.py [hunk 4]
```diff
@@ -1,53 +1,87 @@
...
+            existing_hours_for_task = sum(
+                self.search([('date', '=', date),
+                             ('task_id', '=', task_id)]).mapped('unit_amount'))
+            total_hours_for_day = (total_hours_for_day -
+                                   existing_hours_for_task +
+                                   change)
+        else:
+            total_hours_for_day += change
+
+        if total_hours_for_day > 24:
```
[the remark this answers is not in the document]
Comment (RajatMCAF) [reply mo_hr_timesheets/models/custom_grid_timesheet.py]: I updated

**Review by MORPSmcaf — APPROVED** [review]

**Pull request merged**